packages/@headlessui-vue/src/components/combobox/combobox.ts

```diff
@@ -779,9 +779,10 @@ export let ComboboxLabel = defineComponent({
   name: 'ComboboxLabel',
   props: {
     as: { type: [Object, String], default: 'label' },
-    id: { type: String, default: () => `headlessui-combobox-label-${useId()}` },
+    id: { type: String, default: null },
   },
   setup(props, { attrs, slots }) {
+    let id = props.id ?? `headlessui-combobox-label-${useId()}`
```

Comment:
I could be wrong, but in my testing this shouldn't be required. It looks like the `default()` still runs in the correct context.

In the playground:
<img width="1162" alt="image" src="https://github.com/tailwindlabs/headlessui/assets/1834413/3fbc7d68-6d12-4562-88cf-199df93be054">

In the library:
<img width="1022" alt="image" src="https://github.com/tailwindlabs/headlessui/assets/1834413/60b8134b-127f-46d2-9978-04ce92dc71a7">

In the DOM:
<img width="406" alt="image" src="https://github.com/tailwindlabs/headlessui/assets/1834413/249498d8-0594-4126-b8aa-b14d26e0db22">

Reply:
In Nuxt you get SSR issues when its inside default.